pkg/errors/errors_test.go

```diff
@@ -23,3 +38,2006 @@ func TestErrorFind(t *testing.T) {
+				httpCode: http.StatusUnauthorized,
+				reason:   unauthorizedReason,
+			},
+			want: Unauthorized(unauthorizedReason),
+		},
+		{
+			name: "should return forbidden error",
+			args: args{
+				httpCode: http.StatusForbidden,
+				reason:   forbiddenReason,
+			},
+			want: Forbidden(forbiddenReason),
+		},
+		{
+			name: "should return not found error",
+			args: args{
+				httpCode: http.StatusNotFound,
+				reason:   notFoundReason,
+			},
+			want: NotFound(notFoundReason),
+		},
+		{
+			name: "should return not implemented error",
+			args: args{
+				httpCode: http.StatusMethodNotAllowed,
+				reason:   notImplementedReason,
+			},
+			want: NotImplemented(notImplementedReason),
+		},
+		{
+			name: "should return conflict error",
+			args: args{
+				httpCode: http.StatusConflict,
+				reason:   conflictReason,
+			},
+			want: Conflict(conflictReason),
+		},
+		{
+			name: "should return general error",
+			args: args{
+				httpCode: http.StatusInternalServerError,
+				reason:   generalReason,
+			},
+			want: GeneralError(generalReason),
+		},
+		{
+			name: "general case, should return general error",
+			want: GeneralError("Unspecified error"),
+		},
+	}
+
+	RegisterTestingT(t)
+
+	for _, testcase := range tests {
+		tt := testcase
+		t.Run(tt.name, func(t *testing.T) {
+			t.Parallel()
+			Expect(NewErrorFromHTTPStatusCode(tt.args.httpCode, tt.args.reason)).To(Equal(tt.want))
+		})
+	}
+}
+
+type errorWithoutStackTrace struct {
+}
+
+func (e *errorWithoutStackTrace) Error() string {
+	return "Error"
+}
+func Test_NewWithCause(t *testing.T) {
+	internalServerCause := "Unspecified error"
+	type args struct {
+		code   ServiceErrorCode
+		cause  error
+		reason string
+	}
+	tests := []struct {
+		name string
+		args args
+		want *ServiceError
+	}{
+		{
+			name: "should return a service error with a nil cause",
+			args: args{
+				reason: internalServerCause,
+			},
+			want: &ServiceError{ErrorGeneral, "Unspecified error", http.StatusInternalServerError, nil},
+		},
+		{
+			name: "should return a service error if the cause is not nil",
+			args: args{
+				reason: internalServerCause,
+				cause:  GeneralError(genericErrorMessage),
+			},
+			want: &ServiceError{ErrorGeneral, "Unspecified error", http.StatusInternalServerError, GeneralError("")},
+		},
+		{
+			name: "should return a service error where there is no stack trace",
+			args: args{
+				reason: internalServerCause,
+				cause:  &errorWithoutStackTrace{},
+			},
+			want: &ServiceError{ErrorGeneral, "Unspecified error", http.StatusInternalServerError, errors.WithStack(&errorWithoutStackTrace{})},
+		},
+	}
+
+	RegisterTestingT(t)
+
+	for _, testcase := range tests {
+		tt := testcase
+		t.Run(tt.name, func(t *testing.T) {
+			t.Parallel()
+			err := NewWithCause(tt.args.code, tt.args.cause, tt.args.reason)
+			Expect(err.Code).To(Equal(tt.want.Code))
+			Expect(err.Reason).To(Equal(tt.want.Reason))
+			Expect(err.HttpCode).To(Equal(tt.want.HttpCode))
```

Comment:
I think these 3 lines can be replaced with
```go
Expect(err).to(MatchError(tt.want))
```

Reply:
> I think these 3 lines can be replaced with
> 
> ```go
> Expect(err).to(MatchError(tt.want))
> ```
> 
> see https://onsi.github.io/gomega/#matcherrorexpected-interface

Thanks @ziccardi ! Suggested changes in this example actually result in an error, but I have applied the changes in the cases where it doesn't. Thank you again.